### PR TITLE
Add a note about `journalflags` options to `systemd-journal` backend

### LINKED ISSUE #2696: fail2ban-regex does not find logs from services running as unprivileged users
### Environment:
- Fail2Ban version (including any possible distribution suffixes): Fail2Ban v0.11.1
- OS, including release name/version: Arch Linux, 5.6.3
- [X] Fail2Ban installed via OS/distribution mechanisms
- [X] You have not applied any additional foreign patches to the codebase
- [X] Some customizations were done to the configuration (provide details below is so)

### The issue:

Using the `systemd-journal` backend, `journalmatch` seems to be getting the wrong logs for my service file.

Querying journalctl directly, I see plenty of my tests that I'm hoping to catch (`401 POST`).

```console
$ journalctl _SYSTEMD_UNIT=ml.service --grep '401 POST' -n 20
-- Logs begin at Fri 2019-10-04 15:41:24 MDT, end at Sat 2020-04-18 10:28:35 MDT. --
Apr 18 07:14:25 mymachine python[1029]: [W 07:14:25.805 LabApp] 401 POST /login?next=%2Flab%3F (127.0.0.1) 1.32ms referer=https://[REDACTED]/login?next=%2Flab%3F
Apr 18 07:16:33 mymachine python[1029]: [W 07:16:33.988 LabApp] 401 POST /login?next=%2Flab%3F (127.0.0.1) 1.39ms referer=https://[REDACTED]/login?next=%2Flab%3F
Apr 18 07:18:55 mymachine python[1029]: [W 07:18:55.339 LabApp] 401 POST /login?next=%2Flab%3F (127.0.0.1) 1.33ms referer=https://[REDACTED]/login?next=%2Flab%3F
Apr 18 07:18:59 mymachine python[1029]: [W 07:18:59.130 LabApp] 401 POST /login?next=%2Flab%3F (127.0.0.1) 1.40ms referer=https://[REDACTED]/login?next=%2Flab%3F
Apr 18 10:15:47 mymachine python[1029]: [W 10:15:47.176 LabApp] 401 POST /login?next=%2Flab%3F (192.168.0.115) 1.49ms referer=https://mymachine.local:[REDACTED]/login?next=%2Flab%3F
```

Using `fail2ban-regex`, I am seeing completely different logs in `--print-all-missed` -- so it seems like it isn't even looking at the right systemd unit?

```console
$ sudo fail2ban-regex --journalmatch='_SYSTEMD_UNIT=ml.service' systemd-journal '.*401.*<HOST>.*' --print-all-missed
Running tests
=============

Use   failregex line : .*401.*<HOST>.*
Use         systemd journal
Use         encoding : UTF-8
Use    journal match : _SYSTEMD_UNIT=ml.service


Results
=======

Failregex: 0 total

Ignoreregex: 0 total

Lines: 20 lines, 0 ignored, 0 matched, 20 missed
[processed in 0.00 sec]

|- Missed line(s):
|  2020-04-15T08:38:09.182740mymachine sudoedit[30446]: pam_unix(sudo:session): session opened for user root by (uid=0)
|  2020-04-15T08:38:31.154128mymachine sudoedit[30446]: pam_unix(sudo:session): session closed for user root
|  2020-04-15T08:38:43.311124mymachine sudo[30499]: pam_unix(sudo:session): session opened for user root by (uid=0)
|  2020-04-15T08:38:43.467930mymachine sudo[30499]: pam_unix(sudo:session): session closed for user root
`-
```

#### Expected behavior

I expect `sudo fail2ban-regex --journalmatch='_SYSTEMD_UNIT=ml.service'` to be showing me the same logs as `journalctl _SYSTEMD_UNIT=ml.service`.

#### Observed behavior

I'm seeing entirely different logs.

### PR DESCRIPTION
**Pull request by n8henrie**
Minor documentation addition. Closes #2696

Before submitting your PR, please review the following checklist:

- [x] **CHOOSE CORRECT BRANCH**: if filing a bugfix/enhancement
      against certain release version, choose `0.9`, `0.10` or `0.11` branch,
      for dev-edition use `master` branch
- [ ] **CONSIDER adding a unit test** if your PR resolves an issue
- [X] **LIST ISSUES** this PR resolves
- [ ] **MAKE SURE** this PR doesn't break existing tests
- [ ] **KEEP PR small** so it could be easily reviewed.
- [ ] **AVOID** making unnecessary stylistic changes in unrelated code
- [ ] **ACCOMPANY** each new `failregex` for filter `X` with sample log lines
      within `fail2ban/tests/files/logs/X` file

### CODE REVIEW

File: fail2ban/client/fail2banregex.py [hunk 1]
```diff
@@ -113,7 +113,9 @@ def format_help(self, *args, **kwargs):
 LOG:
     string                  a string representing a log line
     filename                path to a log file (/var/log/auth.log)
-    "systemd-journal"       search systemd journal (systemd-python required)
+    "systemd-journal"       search systemd journal. Optionally specify
+                            `systemd-journal[journalflags=X]` to determine
```
Comment (sebres) [inline fail2ban/client/fail2banregex.py:117]: Basically systemd backend has more parameters as just `journalflags`, see #1523 
Also I must admit, this is currently not documented in [man/jail.conf.5](https://github.com/fail2ban/fail2ban/blob/master/man/jail.conf.5#L300).
Would you possibly like to add that info also?

Comment (n8henrie) [reply fail2ban/client/fail2banregex.py:117]: Thanks for the feedback, see what you think.

**Comment by n8henrie** [comment]
There hasn't been any conversation on this in a little while -- just
wanting to make sure that there are no additional changes I should make to
expedite the process. (In other words, "you're not waiting on me, are you?")

File: fail2ban/client/fail2banregex.py (hunk 1 above)
Comment (sebres) [reply fail2ban/client/fail2banregex.py:117]: > you're not waiting on me, are you?

I thought we were in agreement about to extend this with short note about more parameters than just journalflags... or just a reference to jail.conf.5 which explaining now all that.
Although it is good enough too, so If you'd not extend that, I'll merge it later as it is.

Comment (sebres) [reply fail2ban/client/fail2banregex.py:117]: BTW, the notice "(systemd-python required)" rather belongs to "search systemd journal" as to the second sentence explaining options.

Comment (n8henrie) [reply fail2ban/client/fail2banregex.py:117]: Sorry, I may have misunderstood -- I thought you were asking me to add that info to `jail.conf.5`, so I did: https://github.com/fail2ban/fail2ban/pull/2809/commits/1b7cd75cc15c7f86f63f9c419de82b1b18291e0b

**Comment by sebres** [comment]
OK, I rebased this PR it to 0.10 branch and merged into current 0.11 and 1.0 (master) after small amendment from my side.
Thank you for the contribution!

**Pull request closed without merging**